### PR TITLE
test: add bats tests for csi-secrets-store-provider-alibabacloud

### PR DESCRIPTION
**Pull request by DahuK**
**What type of PR is this?**
/kind failing-test

**What this PR does / why we need it**:
add bats test for a new [provider](https://github.com/AliyunContainerService/secrets-store-csi-driver-provider-alibaba-cloud) from Alibaba Cloud

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
Only the bats tests and testing yaml in this PR, please let me know what should I also prepare like the testing account AK,  cluster or something else? Thanks a lot!

**TODOs**:

- [X] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

### CODE REVIEW

**Comment by aramase** [comment]
/retitle test: add bats tests for csi-secrets-store-provider-alibabacloud

**Comment by DahuK** [comment]
@aramase I have refined this to pass our AK/SK from external secret and pass it into mount request from nodePublishSecretRef, please take your time to review this, thanks!

File: Makefile [hunk 1]
```diff
@@ -214,6 +215,9 @@ $(PROTOC): ## Install protoc
 $(YQ): ## Install yq for running the tests
 	curl -LO https://github.com/mikefarah/yq/releases/download/$(YQ_VERSION)/yq_linux_amd64 && chmod +x ./yq_linux_amd64 && mv yq_linux_amd64 /usr/local/bin/yq
 
+$(ALIYUNCLI): ## Install aliyun for running the tests
+	curl -LO https://github.com/aliyun/aliyun-cli/releases/download/v3.0.134/aliyun-cli-linux-3.0.134-amd64.tgz && tar xzvf aliyun-cli-linux-3.0.134-amd64.tgz && chmod +x ./aliyun && cp aliyun /usr/local/bin
```
Comment (hixichen) [inline Makefile:219]: is that possible to always test with latest aliyuncli?

File: test/bats/tests/alibabacloud/pod-inline-volume-secretproviderclass.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+kind: Pod
+apiVersion: v1
+metadata:
+  name: basic-test-mount
```
Comment (hixichen) [inline test/bats/tests/alibabacloud/pod-inline-volume-secretproviderclass.yaml:4]: nit: aliyun-basic-test-mount?

File: test/bats/alibabacloud.bats [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+setup() {
+  if [[ -z "${ALIBABACLOUD_ACCESS_KEY}" ]] || [[ -z "${ALIBABACLOUD_ACCESS_SECRET}" ]]; then
+    echo "Error: ram ak/sk is not provided" >&2
+    return 1
+  fi
+}
+
+setup_file() {
+    #Configure aliyun cli profile
+    aliyun configure set --profile akProfile --mode AK --region cn-hongkong --access-key-id ${ALIBABACLOUD_ACCESS_KEY} --access-key-secret ${ALIBABACLOUD_ACCESS_SECRET}
```
Comment (hixichen) [inline test/bats/alibabacloud.bats:20]: any specific reason to set `cn-hongkong` as default region?

File: test/bats/tests/alibabacloud/pod-inline-volume-secretproviderclass.yaml (hunk 2 above)
Comment (DahuK) [reply test/bats/tests/alibabacloud/pod-inline-volume-secretproviderclass.yaml]: have updated to alibabacloud-basic-test-mount

File: test/bats/alibabacloud.bats (hunk 3 above)
Comment (DahuK) [reply test/bats/alibabacloud.bats]: refined to us-west-1 as default region

**Comment by DahuK** [comment]
@aramase @hixichen please check this again when you free, thx!

**Review by hixichen — APPROVED** [review]
LGTM

**Comment by DahuK** [comment]
@aramase @hixichen @ritazh  
this has held for a long time, could you please take a time for reviewing this and help to move forward, thanks!

File: Makefile [hunk 4]
```diff
@@ -489,6 +493,10 @@ e2e-akeyless:
 e2e-gcp:
 	bats -t test/bats/gcp.bats
 
+.PHONY: e2e-alibabacloud
+e2e-alibabacloud:
```
Comment (aramase) [inline Makefile:497]: Add `ALIYUNCLI` as a dependency for this target

File: test/bats/alibabacloud.bats [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+
+    result=$(kubectl --namespace=$NAMESPACE get secret $K8_SECRET_NAME -o jsonpath="{.data.username}" | base64 -d)
+    [[ "$result" == $USERNAME ]]
+
+    result=$(kubectl --namespace=$NAMESPACE get secret $K8_SECRET_NAME -o jsonpath="{.data.password}" | base64 -d)
+    [[ "$result" == $PASSWORD ]]
+}
+
+@test "Install alibabacloud provider" {
+    #Install csi secret driver
```
Comment (aramase) [inline test/bats/alibabacloud.bats:57]:
```suggestion
```

File: test/bats/alibabacloud.bats [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+    helm repo add csi-secrets-store-provider-alibabacloud https://raw.githubusercontent.com/AliyunContainerService/secrets-store-csi-driver-provider-alibaba-cloud/main/charts
+    helm install csi-secrets-store-provider-alibabacloud/csi-secrets-store-provider-alibabacloud --namespace kube-system --generate-name --set secrets-store-csi-driver.enableSecretRotation=true --set secrets-store-csi-driver.rotationPollInterval=10s  \
```
Comment (aramase) [inline test/bats/alibabacloud.bats:59]: Secrets Store CSI Driver is already installed with 

https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/350617024bf9c5f5565f7278e16c02185091aabf/Makefile#L421-L423

The provider shouldn't install the driver again as we want to validate the driver built + helm charts as part of the pull request.

File: test/bats/alibabacloud.bats [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+   [[ "${result//$'\r'}" == "afterRotation" ]]
+}
+
+@test "CSI inline volume test with pod portability - read secrets manager secrets from pod" {
+    result=$(kubectl --namespace $NAMESPACE exec $POD_NAME -- cat /mnt/secrets-store/testBasic)
+    [[ "${result//$'\r'}" == "testValue" ]]
+}
+
+@test "CSI inline volume test with pod portability - specify jsmePath for Secrets Manager secret with rotation" {
+
```
Comment (aramase) [inline test/bats/alibabacloud.bats:121]:
```suggestion
```

File: test/bats/alibabacloud.bats [hunk 8]
```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env bats
```
Comment (aramase) [inline test/bats/alibabacloud.bats:1]: Checkout https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/main/test/bats/akeyless.bats which was added recently. I think we only need basic tests for validating the provider works the defined RPC interface and don't need extensive tests like rotation or sync as k8s secrets. The feature specific tests for the driver are already run with e2e-provider and doesn't need to be part of every provider.

File: test/bats/tests/alibabacloud/pod-inline-volume-secretproviderclass.yaml [hunk 9]
```diff
@@ -0,0 +1,25 @@
+kind: Pod
+apiVersion: v1
+metadata:
+  name: alibabacloud-basic-test-mount
+spec:
+  containers:
+    - image: k8s.gcr.io/e2e-test-images/busybox:1.29
```
Comment (aramase) [inline test/bats/tests/alibabacloud/pod-inline-volume-secretproviderclass.yaml:7]:
```suggestion
    - image: registry.k8s.io/e2e-test-images/busybox:1.29-4
```

**Review by aramase — CHANGES_REQUESTED** [review]

File: Makefile (hunk 4 above)
Comment (DahuK) [reply Makefile]: > Add `ALIYUNCLI` as a dependency for this target

updated, thanks!

File: test/bats/alibabacloud.bats (hunk 5 above)
Comment (DahuK) [reply test/bats/alibabacloud.bats]: have fixed

File: test/bats/alibabacloud.bats (hunk 8 above)
Comment (DahuK) [reply test/bats/alibabacloud.bats:1]: Have removed the extensive cases, and only basic test are retained, please review again.

File: test/bats/tests/alibabacloud/pod-inline-volume-secretproviderclass.yaml (hunk 9 above)
Comment (DahuK) [reply test/bats/tests/alibabacloud/pod-inline-volume-secretproviderclass.yaml]: > 

fixed

File: test/bats/alibabacloud.bats (hunk 6 above)
Comment (DahuK) [reply test/bats/alibabacloud.bats]: fix this

**Comment by DahuK** [comment]
@aramase The MR has been on hold for a long time, my apologies for the delayed update. Please review it and let me know if there's anything I need to add.

**Pull request closed without merging**